app/controllers/spree/admin/public_metadata_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Spree
+  module Admin
+    class PublicMetadataController < BaseController
+      include Spree::Admin::MetadataConcern
+
+      before_action :find_resource
+
+      def add; end
+
+      def update
+        assert_metadata(@object)
+
+        if @object.save!
+          respond_to do |format|
+            format.turbo_stream do
+              render turbo_stream: turbo_stream.replace('metadata_form', partial: 'spree/admin/public_metadata/table', locals: { resource: @object })
+            end
+          end
+        end
+      end
+
+      def delete
+        @object.public_metadata.delete(params[:key].to_sym)
+
+        if @object.save!
+          respond_to do |format|
+            format.turbo_stream do
+              render turbo_stream: turbo_stream.remove("metadata_row_#{params[:key]}")
+            end
+          end
+        end
+      end
+
+      private
+
+      def find_resource
```

Comment:
@MatthewKennedy rather than having a separate controller let's add these actions into a concern and include in specific controllers for resources using metadata - this will automatically apply any scope/security concerns existing today